.circleci/test-deploy.yml

```diff
@@ -9,41 +9,91 @@ filters: &filters
   tags:
     only: /.*/
 
+commands:
+  scan:
+    description: test the scan functionality
+    parameters:
+      no-cache:
+        type: boolean
+        default: false
+      test-param-fail-on-no-cache-used:
+        type: boolean
+        default: false
+    steps:
+        - node/install
+        - run:
+            name: Setup test repository
+            command: |
+              git clone https://github.com/snyk-labs/nodejs-goof || echo "Repository already exists, no cloning needed."
+              cd nodejs-goof
+              npm install
+        - snyk/scan:
+            fail-on-issues: false
+            monitor-on-build: false
+            target-file: nodejs-goof/package.json
+            no-cache: << parameters.no-cache >>
+            test-param-fail-on-no-cache-used: << parameters.test-param-fail-on-no-cache-used >>
+
 jobs:
-  installation-test:
+  scan-test:
     docker:
       - image: cimg/base:current
     steps:
-      - checkout
-      - snyk/install
-  scan-test:
+      - scan
+
+  scan-test-ensure-cache-used:
     docker:
       - image: cimg/base:current
     steps:
-      - node/install
-      - run:
-          name: Setup test repository
-          command: |
-            git clone https://github.com/snyk-labs/nodejs-goof
-            cd nodejs-goof
-            npm install
-      - snyk/scan:
-          fail-on-issues: false
-          monitor-on-build: false
-          target-file: nodejs-goof/package.json
+      - scan
+      - scan:
+          test-param-fail-on-no-cache-used: true
+          no-cache: false
 
+  scan-test-old-curl:
+    docker:
+      - image: cimg/base:stable
```

Comment:
Suggestion: since stable will also change at some point, it would make sense to refer to a pinned version instead of an alias